WMI Exporter won't run if Hyper-V collector is enabled

Hi.

I'm trying to run this on a Windows 2012R2 Server which has Hyper-V and failover clustering installed however it won't run. I get the below.

Any ideas? It installs and runs fine on the other cluster node. It also runs fine if I disable the Hyper-V collector.

Many thanks.

Regards
James

C:\Windows\system32>"c:\Program Files\wmi_exporter\wmi_exporter.exe" --log.format logger:eventlog?name=wmi_exporter --collectors.enabled cpu,cs,hyperv,logical_disk,net,os,system --telemetry.addr :9182
←[33mWARN←[0m[0000] Couldn't open registry to determine IIS version:The system cannot find the file specified.  ←[33msource←[0m="iis.go:34"
←[36mINFO←[0m[0000] Enabled collectors: cs, hyperv, logical_disk, net, os, system, cpu  ←[36msource←[0m="exporter.go:214"
←[36mINFO←[0m[0000] Starting WMI exporter (version=0.3.1, branch=master, revision=4b3d1d60d920a2b97de869485e0a4f4162f3ec0e)  ←[36msource←[0m="exporter.go:225"
←[36mINFO←[0m[0000] Build context (go=go1.10, user=appvyr-win\appveyor@APPVYR-WIN, date=20180418-05:52:44)  ←[36msource←[0m="exporter.go:226"
←[36mINFO←[0m[0000] Starting server on :9182                      ←[36msource←[0m="exporter.go:229"
panic: inconsistent label cardinality

goroutine 46 [running]:
github.com/martinlindhe/wmi_exporter/vendor/github.com/prometheus/client_golang/prometheus.MustNewConstMetric(0xc0421f65b0, 0x2, 0x0, 0xc0422f5ea0, 0x1, 0x1, 0x
0, 0x0)
        c:/gopath/src/github.com/martinlindhe/wmi_exporter/vendor/github.com/prometheus/client_golang/prometheus/value.go:172 +0xb7
github.com/martinlindhe/wmi_exporter/collector.(*HyperVCollector).collectVmEthernet(0xc0421f8000, 0xc042156600, 0x0, 0x0, 0x0)
        c:/gopath/src/github.com/martinlindhe/wmi_exporter/collector/hyperv.go:1125 +0x20e
github.com/martinlindhe/wmi_exporter/collector.(*HyperVCollector).Collect(0xc0421f8000, 0xc042156600, 0xba1540, 0x0)
        c:/gopath/src/github.com/martinlindhe/wmi_exporter/collector/hyperv.go:540 +0x76e
main.execute(0xc0420a8097, 0x6, 0x95b080, 0xc0421f8000, 0xc042156600)
        c:/gopath/src/github.com/martinlindhe/wmi_exporter/exporter.go:84 +0x85
main.WmiCollector.Collect.func1(0xc042156600, 0xc042172430, 0xc0420a8097, 0x6, 0x95b080, 0xc0421f8000)
        c:/gopath/src/github.com/martinlindhe/wmi_exporter/exporter.go:64 +0x5a
created by main.WmiCollector.Collect
        c:/gopath/src/github.com/martinlindhe/wmi_exporter/exporter.go:63 +0x110

--- a/collector/hyperv.go
+++ b/collector/hyperv.go
@@ -463,37 +463,37 @@ func NewHyperVCollector() (Collector, error) {
 		AdapterBytesDropped: prometheus.NewDesc(
 			prometheus.BuildFQName(Namespace, buildSubsystemName("ethernet"), "bytes_dropped"),
 			"Bytes Dropped is the number of bytes dropped on the network adapter",
-			nil,
+			[]string{"adapter"},
 			nil,
 		),
 		AdapterBytesReceived: prometheus.NewDesc(
 			prometheus.BuildFQName(Namespace, buildSubsystemName("ethernet"), "bytes_received"),
 			"Bytes received is the number of bytes received on the network adapter",
-			nil,
+			[]string{"adapter"},
 			nil,
 		),
 		AdapterBytesSent: prometheus.NewDesc(
 			prometheus.BuildFQName(Namespace, buildSubsystemName("ethernet"), "bytes_sent"),
 			"Bytes sent is the number of bytes sent over the network adapter",
-			nil,
+			[]string{"adapter"},
 			nil,
 		),
 		AdapterFramesDropped: prometheus.NewDesc(
 			prometheus.BuildFQName(Namespace, buildSubsystemName("ethernet"), "frames_dropped"),
 			"Frames Dropped is the number of frames dropped on the network adapter",
-			nil,
+			[]string{"adapter"},
 			nil,
 		),
 		AdapterFramesReceived: prometheus.NewDesc(
 			prometheus.BuildFQName(Namespace, buildSubsystemName("ethernet"), "frames_received"),
 			"Frames received is the number of frames received on the network adapter",
-			nil,
+			[]string{"adapter"},
 			nil,
 		),
 		AdapterFramesSent: prometheus.NewDesc(
 			prometheus.BuildFQName(Namespace, buildSubsystemName("ethernet"), "frames_sent"),
 			"Frames sent is the number of frames sent over the network adapter",
-			nil,
+			[]string{"adapter"},
 			nil,
 		),
 	}, nil

